nncf/experimental/torch/nas/bootstrapNAS/search/search.py

```diff
@@ -193,19 +210,33 @@ def __init__(self,
         self._verbose = verbose
         self._top1_accuracy_validation_fn = None
         self._val_loader = None
-        evo_algo = search_config['algorithm']
-        if evo_algo == EvolutionaryAlgorithms.NSGA2.value:
+        self._algorithm_name = search_config['algorithm']
+        sampling = get_sampling("int_lhs")
+        if self._algorithm_name == EvolutionaryAlgorithms.NSGA2.value:
             self._algorithm = NSGA2(pop_size=self.search_params.population,
-                                    sampling=get_sampling("int_lhs"),
+                                    sampling=sampling,
                                     crossover=get_crossover("int_sbx", prob=self.search_params.crossover_prob,
                                                             eta=self.search_params.crossover_eta),
                                     mutation=get_mutation("int_pm", prob=self.search_params.mutation_prob,
                                                           eta=self.search_params.mutation_eta),
                                     eliminate_duplicates=True,
                                     save_history=True,
                                     )
+        elif self._algorithm_name == EvolutionaryAlgorithms.RNSGA2.value:
+            self._algorithm = RNSGA2(ref_points=self.search_params.aspiration_points,
```

Comment:
Why not just pass “search_params” itself? It became too hard to call a search algorithm from outside by specifying all these parameters. 

Ideally, it should be like this:
```python
search_params_json = json.load(...)
search_params = SearchParams.from_dict(search_params)
search_algo = RNSGAII(search_params, **extra_params)
```

Reply:
Refactored the code based on your suggestion. Thanks, @ljaljushkin !